### PR TITLE
[bitnami/Joomla] Major version. Adapt Chart to apiVersion: v2 and Update MariaDB Dependency

### PR DESCRIPTION
**Pull request by FraPazGal**
**Description of the change**

- Bump `apiVersion` from `v1` to `v2` in _Chart.yaml_
- Fields in _Chart.yaml_ file has been ordered alphabetically
- Move dependency information from the _requirements.yaml_ to the _Chart.yaml_
- After running `helm dependency update`, a new _Chart.lock_ file is generated containing the same structure used in the previous _requirements.lock_
- Update _README.md_
- Updated MariaDB dependency to the new major.
- Added some missing parameters present on `charts/template`
- Added support for existingSecret for external databases

**Possible drawbacks**

- Helm v2 is no longer supported
- Breaking changes due to MariaDB major bump

### CODE REVIEW

File: bitnami/joomla/Chart.yaml [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+apiVersion: v2
 appVersion: 3.9.22
+dependencies:
+- condition: mariadb.enabled
+  name: mariadb
+  repository: https://charts.bitnami.com/bitnami
+  version: 8.x.x
+- name: common
+  repository: https://charts.bitnami.com/bitnami
+  version: 0.x.x
```
Comment (carrodher) [inline bitnami/joomla/Chart.yaml:12]: Here we should use the new common major
```suggestion
  version: 1.x.x
```

File: bitnami/joomla/Chart.yaml [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
-version: 8.1.10
+annotations:
+  category: CMS
+apiVersion: v2
 appVersion: 3.9.22
+dependencies:
+- condition: mariadb.enabled
+  name: mariadb
+  repository: https://charts.bitnami.com/bitnami
+  version: 8.x.x
```
Comment (carrodher) [inline bitnami/joomla/Chart.yaml:9]: Here we should use the new mariadb major
```suggestion
  version: 9.x.x
```

File: bitnami/joomla/Chart.lock [hunk 3]
```diff
@@ -0,0 +1,9 @@
+dependencies:
+- name: mariadb
+  repository: https://charts.bitnami.com/bitnami
+  version: 8.0.7
+- name: common
+  repository: https://charts.bitnami.com/bitnami
+  version: 0.10.0
+digest: sha256:dffbd0242679c5af046f5701d9078fd8efcbfe8f81d2cc4054c200bb0a03b696
+generated: "2020-11-11T20:30:50.056218268Z"
```
Comment (carrodher) [inline bitnami/joomla/Chart.lock:9]: This file is going to be automatically updated by the pipeline once executed the Docker Image job using this PR as parameter

**Review by carrodher — COMMENTED** [review]
I just reviewed only the changes related to Helm 2 EOL, just a couple of comments

File: bitnami/joomla/Chart.yaml [hunk 4]
```diff
@@ -1,22 +1,30 @@
...
+- blog
+- http
+- php
 maintainers:
-  - name: Bitnami
-    email: containers@bitnami.com
-engine: gotpl
-annotations:
-  category: CMS
+- email: containers@bitnami.com
```
Comment (miguelaeh) [inline bitnami/joomla/Chart.yaml:24]: I would say this indentation level will not pass the linter. I should be two spaces. The same happens in all the arrays of this file

File: bitnami/joomla/templates/externaldb-secrets.yaml [hunk 5]
```diff
@@ -1,10 +1,17 @@
-{{- if not .Values.mariadb.enabled }}
+{{- if (not (or .Values.mariadb.enabled .Values.externalDatabase.existingSecret)) }}
```
Comment (miguelaeh) [inline bitnami/joomla/templates/externaldb-secrets.yaml:1]: Not important but I think a `(` level can be removed

File: bitnami/joomla/templates/externaldb-secrets.yaml [hunk 6]
```diff
@@ -1,10 +1,17 @@
...
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
 type: Opaque
 data:
-  db-password: {{ default "" .Values.externalDatabase.password | b64enc | quote }}
+  mariadb-password: {{ default "" .Values.externalDatabase.password | b64enc | quote }}
```
Comment (miguelaeh) [inline bitnami/joomla/templates/externaldb-secrets.yaml:16]: Good catch, I had this as pending stuff!!

**Review by miguelaeh — CHANGES_REQUESTED** [review]

File: bitnami/joomla/Chart.yaml (hunk 4 above)
Comment (carrodher) [reply bitnami/joomla/Chart.yaml:24]: Yep, I think so. Running the following commands everything should be prettified as expected
```
$ yq r --tojson Chart.yaml | jq . | yq r --prettyPrint - > Chart.yaml.new
$ mv Chart.yaml.new Chart.yaml
```

**Pull request merged**